processing/src/main/java/io/druid/query/filter/JavaScriptDimFilter.java

```diff
@@ -21,32 +21,47 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Joiner;
 import com.google.common.base.Preconditions;
 import com.metamx.common.StringUtils;
 
 import java.nio.ByteBuffer;
+import java.util.Arrays;
 
 public class JavaScriptDimFilter implements DimFilter
 {
-  private final String dimension;
+  private final String[] dimensions;
   private final String function;
 
   @JsonCreator
   public JavaScriptDimFilter(
       @JsonProperty("dimension") String dimension,
+      @JsonProperty("dimensions") String[] dimensions,
       @JsonProperty("function") String function
   )
   {
-    Preconditions.checkArgument(dimension != null, "dimension must not be null");
+    Preconditions.checkArgument(dimension != null || dimensions != null, "dimension or dimensions must not be null");
+    Preconditions.checkArgument(!(dimension != null && dimensions != null), "both dimension and dimensions are defined");
     Preconditions.checkArgument(function != null, "function must not be null");
-    this.dimension = dimension;
+    this.dimensions = (dimension != null) ? new String[]{dimension} : dimensions;
```

Comment:
the new list form should take precedence over the old backwards-compatible one, not the other way around.